dask/base.py

```diff
@@ -1397,14 +1396,16 @@ def get_scheduler(get=None, scheduler=None, collections=None, cls=None):
     if config.get("get", None):
         raise ValueError(get_err_msg)
 
-    if getattr(thread_state, "key", False):
-        from distributed.worker import get_worker
-
-        return get_worker().client.get
```

Comment:
This is equivalent to the get_client call I introduced below if `getattr(thread_state, "key", False)` is True so there is no need to have this additional indirection.

By having only a single call, we're changing precendee slightly s.t. a class providing the `__dask_scheduler__` attribute wins but I believe this "more correct" considering it is passed explicitly. I doubt this has any relevance.